src/components/connections.vue

```diff
@@ -10,18 +10,35 @@ export default {
     return {
       name: connectionName,
       connectionString,
-      isNewConnectionFormVisible: false
+      isNewConnectionFormVisible: false,
+      isConnected: null
     }
   },
   methods: {
     ...mapActions(['connect', 'setDatabase', 'setCollection']),
     async doConnect() {
       await this.connect(this.connectionString)
+      this.isConnected = true
       this.$router.push('/connection')
+    },
+    setSelectedDBorCollection(sourceType, sourceName) {
```

Comment:
i believe you don't need this method here. simply returning false in the store in both `setDatabase` and `setCollection` if the names are the same should solve this issue.